sdk/typescript/src/types/events.ts

```diff
@@ -53,3 +54,38 @@ export type SuiEvent =
     | { newObject: NewObjectEvent }
     | { epochChange: bigint }
     | { checkpoint: bigint };
+
+export type MoveEventField = {
+    path: string,
+    value: SuiJsonValue
+}
+
+// mirrors sui_framework::EventType
+export type EventType =
+    | "TransferToAddress"
+    | "TransferToObject"
+    | "FreezeObject"
+    | "ShareObject"
+    | "DeleteObjectID"
+    | "User";
+
+// mirrors sui_json_rpc_types::SuiEventFilter
+export type SuiEventFilter =
+    | { "package" : ObjectId }
+    | { "module" : string }
+    | { "moveEventType" : string }
+    | { "moveEventField" : MoveEventField }
+    | { "senderAddress" : SuiAddress }
+    | { "eventType" : EventType }
+    | { "All" : SuiEventFilter[] }
+    | { "Any" : SuiEventFilter[] }
+    | { "And" : [SuiEventFilter, SuiEventFilter] }
+    | { "Or" : [SuiEventFilter, SuiEventFilter] };
+
+export type SuiEventEnvelope = {
+    timestamp: bigint | number,
```

Comment:
```suggestion
    timestamp: number,
```

don't think timestamp can ever be a bigint